eth-contracts/contracts/EthRewardsManager.sol

```diff
@@ -0,0 +1,168 @@
+        "EthRewardsManager: Only governance"
+    );
+
+    address private governanceAddress;
+
+    /// @dev ERC-20 token that will be used to stake with
+    ERC20 internal audiusToken;
+    Wormhole internal wormhole;
+    bytes32 internal recipient;
+
+    address public antiAbuseOracleAddress;
+
+    /**
+     * @notice Function to initialize the contract
+     * @param _tokenAddress - address of ERC20 token
+     * @param _governanceAddress - address for Governance proxy contract
+     * @param _wormholeAddress - address for Wormhole contract
+     * @param _recipient - solana address of recipient
+     * @param _antiAbuseOracleAddress - address for antiAbuseOracleAddress
+     */
+    function initialize(
+        address _tokenAddress,
+        address _governanceAddress,
+        address _wormholeAddress,
+        bytes32 _recipient,
+        address _antiAbuseOracleAddress
+    ) public initializer {
+        require(Address.isContract(_tokenAddress), ERROR_TOKEN_NOT_CONTRACT);
+        require(
+            Address.isContract(_wormholeAddress),
+            ERROR_WORMHOLE_NOT_CONTRACT
+        );
+        audiusToken = ERC20(_tokenAddress);
+        wormhole = Wormhole(_wormholeAddress);
+        recipient = _recipient;
+        antiAbuseOracleAddress = _antiAbuseOracleAddress;
+        _updateGovernanceAddress(_governanceAddress);
+        InitializableV2.initialize();
+    }
+
+    /**
+     * @notice Set the Governance address
+     * @dev Only callable by Governance address
+     * @param _governanceAddress - address for new Governance contract
+     */
+    function setGovernanceAddress(address _governanceAddress) external {
+        _requireIsInitialized();
+
+        require(msg.sender == governanceAddress, ERROR_ONLY_GOVERNANCE);
+        _updateGovernanceAddress(_governanceAddress);
+    }
+
+    /**
+     * @notice Set the recipient address
+     * @dev Only callable by Governance address
+     * @param _recipient - address for new recipient
+     */
+    function setRecipientAddress(bytes32 _recipient) external {
+        _requireIsInitialized();
+
+        require(msg.sender == governanceAddress, ERROR_ONLY_GOVERNANCE);
+        recipient = _recipient;
+    }
+
+    /**
+     * @notice Set antiAbuseOracleAddress
+     * @dev Only callable by Governance address
+     * @param _antiAbuseOracleAddress - value of new antiAbuseOracleAddress
+     */
+    function setAntiAbuseOracleAddress(address _antiAbuseOracleAddress) external {
+        _requireIsInitialized();
+
+        require(msg.sender == governanceAddress, ERROR_ONLY_GOVERNANCE);
+        antiAbuseOracleAddress = _antiAbuseOracleAddress;
+    }
+
+    /* External functions */
+
+    /**
+     * @notice Transfers to solana
+     * @param _nonce - nonce for wormhole
+     */
+    function transferToSolana(uint32 _nonce) external {
+        _requireIsInitialized();
+
+        require(msg.sender == governanceAddress, ERROR_ONLY_GOVERNANCE);
```

Comment:
i don't think this should always be governance right? since that would require a proposal and voting process each time